Tella/Data/Crypto/Cryptor.swift

```diff
@@ -0,0 +1,161 @@
+//
+//  Cryptor.swift
+//  Tella
+//
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import CommonCrypto
+
+// Enum to handle encryption errors
+enum CryptoError: Error {
+    case cryptorCreationFailed
+    case encryptionFailed
+    case finalizationFailed
+}
+
+class Cryptor {
+    
+    static let bufferSize = 1024 * 1024 * 8  // 8 MB buffer size
+    
+    private var cryptor: CCCryptorRef?
+    private var cryptoOperation: CryptoOperationEnum
+    private var inputFile:FileHandle
+    private var outputFile:FileHandle
+    private var encryptionKeyData: Data
+    
+    init(inputFileURL: URL,
+         outputFileURL:URL,
+         encryptionKeyData: Data,
+         cryptoOperation: CryptoOperationEnum) throws {
+        
+        let _ = inputFileURL.startAccessingSecurityScopedResource()
+        defer { inputFileURL.stopAccessingSecurityScopedResource() }
+        
+        let _ = outputFileURL.startAccessingSecurityScopedResource()
+        defer { outputFileURL.stopAccessingSecurityScopedResource() }
+        
+        // Open the input and output file handles
+        self.inputFile = try FileHandle(forReadingFrom: inputFileURL)
+        self.outputFile = try FileHandle(forWritingTo: outputFileURL)
+        
+        self.cryptoOperation = cryptoOperation
+        self.encryptionKeyData = encryptionKeyData
+    }
+    
+    func cryptFile() throws {
+        try cryptorCreate()
+        try cryptorUpdate()
+        try cryptorFinal()
+    }
+    
+    private func cryptorCreate() throws  {
+        // Setup the encryption context
+        let keySize = size_t(kCCKeySizeAES256)
+        let algorithm = CCAlgorithm(kCCAlgorithmAES)
+        let options = CCOptions(ccNoPadding)
+        let mode = CCMode(kCCModeCTR)
+        let operation: CCOperation = cryptoOperation == .encrypt ? UInt32(kCCEncrypt) : UInt32(kCCDecrypt)
+        
+        let operationResult = encryptionKeyData.withUnsafeBytes { keyBytes in
+            
+            CCCryptorCreateWithMode(
+                operation,
+                algorithm,
+                mode,
+                options,
+                nil, // Initialization Vector
+                keyBytes.baseAddress, // Encryption Key
+                keySize,
+                nil,
+                0,
+                0,
+                CCModeOptions(kCCModeOptionCTR_BE),
+                &cryptor
+            )
+        }
+        
+        if operationResult !=  kCCSuccess {
+            debugLog("Error encryption \(operationResult)", space: .crypto)
+            throw CryptoError.cryptorCreationFailed
+        }
+    }
+    
+    private func cryptorUpdate() throws {
+        var stop = false
+        
```

Comment:
could we add a defer block to close the inputFile and outputFile when the method exits? I know we are doing this inside of the while loop but this way the method is more readable

Reply:
I think we need to put the `defer` in `cryptFile()` function because we still use  `inputFile` and `outputFile` in `cryptorFinal()` function